integration-tests/tests/src/tests/results.ts

```diff
@@ -0,0 +1,1067 @@
+      expect(() => {
+        results.sum("foo");
+      }).throws("Property 'foo' does not exist on object 'NullableBasicTypesObject'");
+      expect(() => {
+        results.avg("foo");
+      }).throws("Property 'foo' does not exist on object 'NullableBasicTypesObject'");
+    });
+  });
+  describe("Updating Results", () => {
+    openRealmBeforeEach({ schema: [NullableBasicTypesSchema] });
+    it("should update correctly", function (this: RealmContext) {
+      const N = 5;
+
+      this.realm.write(() => {
+        for (let i = 0; i < N; i++) {
+          this.realm.create("NullableBasicTypesObject", { intCol: 10 });
+        }
+      });
+
+      // update should work on a basic result set
+      let results = this.realm.objects("NullableBasicTypesObject");
+      expect(results.length).equals(5);
+      this.realm.write(() => {
+        results.update("intCol", 20);
+      });
+      expect(results.length).equals(5);
+      expect(this.realm.objects("NullableBasicTypesObject").filtered("intCol = 20").length).equals(5);
+
+      // update should work on a filtered result set
+      results = this.realm.objects("NullableBasicTypesObject").filtered("intCol = 20");
+      this.realm.write(() => {
+        results.update("intCol", 10);
+      });
+      expect(results.length).equals(0);
+      expect(this.realm.objects("NullableBasicTypesObject").filtered("intCol = 10").length).equals(5);
+
+      // update should work on a sorted result set
+      results = this.realm.objects("NullableBasicTypesObject").filtered("intCol == 10").sorted("intCol");
+      this.realm.write(() => {
+        results.update("intCol", 20);
+      });
+      expect(results.length).equals(0);
+      expect(this.realm.objects("NullableBasicTypesObject").filtered("intCol = 20").length).equals(5);
+
+      // update should work on a result snapshot
+      results = this.realm.objects("NullableBasicTypesObject").filtered("intCol == 20").snapshot();
+      this.realm.write(() => {
+        results.update("intCol", 10);
+      });
+      expect(results.length).equals(5); // snapshot length should not change
+      expect(this.realm.objects("NullableBasicTypesObject").filtered("intCol = 10").length).equals(5);
+    });
+
+    it("should update correctly with all types", function (this: RealmContext) {
+      const objectsCreated = 5;
+      /**
+       * We are using BasicTypesObject instead of NullableBasicTypes object to make sure
+       * if the basic types change this test gets updated.
+       */
+      const initialObjectValues: BasicTypesObject = {
+        boolCol: false,
+        stringCol: "hello",
+        intCol: 10,
+        floatCol: 10.0,
+        doubleCol: 10.0,
+        dateCol: new Date(10),
+        dataCol: new Uint8Array([0xd8, 0x21, 0xd6]),
+        decimal128Col: Decimal128.fromString("3e-10"),
+        objectIdCol: new ObjectId(`0000002a9a7969d24bea4cf1`),
+        uuidCol: new UUID(`f7adc509-49be-466f-b3d5-449696d9a601`),
```

Comment:
I refactored the old test to make it more clear and also added some new Realm types to it, might be worth looking at this more.